### PR TITLE
Edit for packaging

### PR DESCRIPTION
**Pull request by hrshdhgd**
- [x] Moved `src` and `project` dir out of the `sssom_schema` dir. So basically reverting back to the autogenerated file structure.
 - [x] Renamed `linkml` to `schema` and moved it under the `datamodel` dir
 
Full explanation: https://github.com/linkml/linkml/issues/797#issuecomment-1147740606

### CODE REVIEW

**Review by matentzn — COMMENTED** [review]
Looks good to me! I fixed one issue with the schema and adjusted the makefile to the new layout.

- all the generated project docs are missing - are these just gitignored?

File: pyproject.toml [hunk 1]
```diff
@@ -1,16 +1,13 @@
 [tool.poetry]
 name = "sssom-schema"
-version = "0.1.3"
+version = "0.0.0"
```
[the remark this answers is not in the document]
Comment (hrshdhgd) [reply pyproject.toml:3]: Yes, GitHub takes care of versioning. Linkml does the same.

**Comment by hrshdhgd** [comment]
Aah ... apparently they were always in `.gitignore`. I did not actively do it. Seem like it has been there for the last 2 months.

**Comment by hrshdhgd** [comment]
If all of them are autogenerated then the GitHub workflow will take care of them, that should not be an issue. If there are files that were manually created, we need to version control them.

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (matentzn) [reply pyproject.toml:3]: thx

**Comment by hrshdhgd** [comment]
Also, we need to decide if the entire `schema` folder is needed or needs to be deleted (or just `sssom_datamodel.py`)

**Pull request merged**